### PR TITLE
Bug 1796939 - Set libthai dictionary path for Firefox and others

### PR DESCRIPTION
**Pull request by lissyx**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----

### CODE REVIEW

File: extensions/desktop/common/desktop-exports [hunk 1]
```diff
@@ -467,3 +467,6 @@ ln -sfn "$REALHOME/.config/ibus/bus" "$IBUS_CONFIG_PATH"
 
 # Set libgweather path
 export LIBGWEATHER_LOCATIONS_PATH="$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/libgweather-4/Locations.bin"
+
+# Set libthai dict path
+export LIBTHAI_DICTDIR="$SNAP/gnome-platform/usr/share/libthai/"
```
Comment (peat-psuwit) [inline extensions/desktop/common/desktop-exports:472]: Hmm... shouldn't this need to be `$SNAP_DESKTOP_RUNTIME` instead, in case the runtime used is KDE?

Comment (lissyx) [reply extensions/desktop/common/desktop-exports:472]: so `$SNAP/$SNAP_DESKTOP_RUNTIME/usr/share/libthai/` ?

Comment (peat-psuwit) [reply extensions/desktop/common/desktop-exports:472]: Judging from the line above, I think it's more likely just `$SNAP_DESKTOP_RUNTIME/usr/share/libthai/`.

Note that it's not as important for KDE runtime, as Qt seems to use ICU for Thai word breaking. However, KDE runtime _does_ include libthai, so it's possible that snap'ed application uses libthai from KDE runtime and run into the same problem.

**Review by kenvandine — APPROVED** [review]
Thanks for the fix, this is better

**Comment by lissyx** [comment]
Is there a problem with CI ? My first push was at 0900 this morning, and it never ran, not even `cla-check` :(

**Comment by lissyx** [comment]
@kenvandine looks like CI was successfull in the end, I guess it's ready to merge :)

**Comment by lissyx** [comment]
@kenvandine @peat-psuwit Can we merge this ? Thanks !

**Comment by peat-psuwit** [comment]
I'm OK with this PR now. But I'm not a maintainer but rather the original reporter. 😅

**Comment by sergiusens** [comment]
Hi, if @kenvandine is fine with this, so am I; but if we want this in the next release (this coming Monday), I would kindly ask if you can retarget (and rebase) this branch against hotfix/7.4

**Comment by kenvandine** [comment]
Fine with me, it would be good to land this.

**Comment by sergiusens** [comment]
I will wait for the rebase then. Thanks

**Comment by lissyx** [comment]
somehow missed the github notification but rebased

**Comment by lissyx** [comment]
> Hi, if @kenvandine is fine with this, so am I; but if we want this in the next release (this coming Monday), I would kindly ask if you can retarget (and rebase) this branch against hotfix/7.4

https://github.com/snapcore/snapcraft/pull/4199

**Comment by sergiusens** [comment]
closing this as https://github.com/snapcore/snapcraft/pull/4199 is coming to main soon

**Pull request closed without merging**